frontend/src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Breadcrumb, Layout, Menu } from 'antd';
+import styles from './Menu.module.css';
+import { items } from '../../consts/MenuItem.tsx';
+
+const { Header, Content, Footer, Sider } = Layout;
+
+const LeftMenu: React.FC = () => {
+    const [collapsed, setCollapsed] = useState(false);
+    const [currentPage, setCurrentPage] = useState('');
+    const [selectedKey, setSelectedKey] = useState('1');
+
+    const findItemByKey = (key, items) => {
+        for (const item of items) {
+            if (item.key === key) {
+                return item;
+            }
+            // If the item has children, search recursively in children
+            if (item.children) {
+                const found = findItemByKey(key, item.children);
+                if (found) {
+                    return {
+                        sectionLabel: item.label,
+                        childLabel: found.label,
+                    };
+                }
+            }
+        }
+        return null;
+    };
+
+    const setCurrentPageFromItem = (item) => {
+        if (item.childLabel) {
+            setCurrentPage(`${item.sectionLabel}/${item.childLabel}`);
+        } else {
+            setCurrentPage(item.label);
+        }
+    };
+
+    const handleMenuClick = (e) => {
+        setSelectedKey(e.key);
+        const selectedItem = findItemByKey(e.key, items);
+
+        if (selectedItem) {
+            setCurrentPageFromItem(selectedItem); // Call the new function here
+        } else {
+            setCurrentPage(''); // Reset currentPage if no item is selected
+        }
+    };
+
+// Effect to set the default page title based on the initially selected key
+    useEffect(() => {
+        const defaultItem = findItemByKey(selectedKey, items);
+        if (defaultItem) {
+            setCurrentPageFromItem(defaultItem); // Call the new function here as well
+        }
+    }, []); // Runs only once after the component mounts
+
+
+    return (
+        <Layout className={styles.menuLayout}>
+            <Sider
+                className={styles.menuSider}
+                collapsible
+                collapsed={collapsed}
+                onCollapse={(value) => setCollapsed(value)}
+            >
+                <div className={styles.logo}>
+                    <span className={styles.logoText}>k8_user</span>
+                </div>
```

Comment:
If this is where user name will be placed, there should be some form of length check or hiding rest of name when it is too long.
<div className={styles.logo}>
                    <span className={styles.logoText}>grzegorz_brzęczyszczykiewicz@pwr.edu.pl</span>
                </div>
![image](https://github.com/user-attachments/assets/1d708f0c-3c32-4702-b321-cc3872c01a64)